pkg/overlay/cache.go

```diff
@@ -31,6 +33,27 @@ type Cache struct {
 	DHT dht.DHT
 }
 
+// NewRedisOverlayCache returns a pointer to a new Cache instance with an initialized connection to Redis.
+func NewRedisOverlayCache(address, password string, dbindex int, dht dht.DHT) (*Cache, error) {
```

Comment:
Don't add these. The setup should be done in config setup, not here.

Reply:
from the merge in master, I've cleaned them up